storage/delegatelayerhandler.go

```diff
@@ -41,19 +41,28 @@ func newDelegateLayerHandler(storageDriver storagedriver.StorageDriver, options
 // Resolve returns an http.Handler which can serve the contents of the given
 // Layer, or an error if not supported by the storagedriver.
 func (lh *delegateLayerHandler) Resolve(layer Layer) (http.Handler, error) {
-	layerURL, err := lh.urlFor(layer)
+	// TODO(bbland): This is just a sanity check to ensure that the
+	// storagedriver supports url generation. It would be nice if we didn't have
+	// to do this twice.
+	_, err := lh.urlFor(layer, map[string]interface{}{"method": "GET"})
 	if err != nil {
 		return nil, err
 	}
 
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		layerURL, err := lh.urlFor(layer, map[string]interface{}{"method": r.Method})
```

Comment:
You only need to do this if the method is not a get. So, save the url from above in the closure and then check if the request method is different.

;)